### ISSUE TITLE
Bad accessibility practice in SC1.4.13 Understanding document

### ISSUE THREAD

**Issue by guyhickling** [w3c/wcag21#808]
I have just noticed a serious accessibility mistake in the Understanding document for SC1.4.13, in the 30 Jan 2018 release of 2.1. I am sure it is just an oversight and not intended, but it will cause trouble for generations of keyboard and screen reader users if it is not deleted from the document now. Changing just two words in the Understanding document will do it. 

Under the "Additional Notes" heading, the third bullet point says: 
"Content which can be triggered via pointer hover should also be able to be triggered by keyboard focus".

This should absolutely not be the case! Developers often do it, but the WCAG should be forbidding it, not encouraging it! For purposes of this discussion lets use one of the most common examples of this kind of mechanism, the "Info" or "Help" button that reveals a small box of text on mouse hover. 

### Different users

The non-disabled mouse user has a choice of whether to hover over the Info button and see the text or not. If they have already read the information once they probably won't want to do so again. 

But if you make the text appear on focus as well, you are withholding that choice from keyboard-only users. These users will have no choice about whether they want to see the text again - every time they navigate past that point in the content using the Tab key, the tooltip will appear once more, providing a repetitive distraction. Even more repetitive when there are several of these things on the page as is often the case. 

This becomes even more of a nuisance for screen reader users whose readers will announce the tooltip each time on its appearance. But screen reader users cannot immediately ignore it, as sighted people can ignore a popup tooltip. They always have to hear the first few words before deciding, "Oh that’s just that annoying info thing again" and press the key to move on. And the first time they arrive at these things they have to listen to most or all of the text, even on subjects they already know about, until they decide it just seems to be extra information with nothing very important in it, and they can skip the rest of it.

In short, keyboard and screen reader users must be given the same choice of whether or not to see or hear such content that non-disabled mouse users have. 

### The solution

For Info and Help buttons that reveal a tooltip, and other similar components including menu options that open a submenu, the developer should only display the hidden content when the keyboard user presses the Enter key while focus is on the button. Do not display it just because the user lands on the button. Since the &lt;button&gt; element is normally the element used for things like Info tooltips (or should be) the Enter and spacebar keys will both be inherently available to display the tooltip text. 

If all this still does not seem right, consider a different component, a set of accordion bars that the mouse user can click to open and reveal content. We would not reveal each one in turn as a keyboard user tabs from one accordion bar to the next, would we? The principle is the same with tooltips, submenus and other similar components. All users, including disabled people, should have the same choice of whether to expand them or not.

### Amendment

Can we please have this Understanding document amended urgently before it becomes final? (This Understanding document needs other changes as well but I have raised a separate issue for that.) The third bullet point should be amended to read:

> Content which can be triggered via pointer hover should also be triggered by pressing the usual keys for operating a focused button.

**Comment by joshueoconnor** [comment]
Thanks for the comment, the working group is aware of this issue and it is actively being worked on by @steverep. We will be re-issuing an updated Understanding doc in due course that will contain the new information.

**Comment by mraccess77** [comment]
Personal comment -- I would be concerned about us forbidden displaying content on focus.   Promoting something is different than banning it and I think there are cases where new content can appear on focus and not cover up anything and it can be useful.  Requiring the user to press enter to display something is not equivalent to the mouse user who only needs to hover and not click.  I generally agree that display things on focus is problematic -- but am against a blanket ban on it.

**Comment by guyhickling** [comment]
Jonathan, thank you for joining in, and I agree that pressing Enter on focus is not the same kind of action physically, but that's the difference between keyboard and mouse use and there's no getting away from that. I do think pressing a key more closely achieves the same end result for users with disabilities. 

Mouse users have a choice to hover or not, so disabled users should have the same choice. Showing things automatically on focus forces the hidden content on them, and takes away that choice. Effectively it discriminates against them even though not intended. So it's bad practice and that's what the WCAG is for. Yes, there may be some edge cases where the opposite might be better; they could be mentioned as exceptions if someone can think of any.

**Comment by mraccess77** [comment]
@guyhickling It's also worth pointing out that when I use the mouse I accidentally trigger hidden content with my pointer frequently.  This proposed SC helps with that by allowing me to dismiss that content with the keyboard -- but I still have to actively do that -- so even as a sometimes mouse user on my way to the search field at the top of a site I may often trigger a navigation menu on hover.  Similarly, I often accidentally close pop-ups that I need to use because my pointer strays out of the bounds.  I can think of situations where showing content on focus will more quickly help the keyboard user such as showing shows from a list of genres without requiring the user press enter.  Focus changes may occurs with arrow keys.  Ultimately additional content popping up can be distracting to users as well -- so there may be no right answer that works for everyone.

**Comment by joshueoconnor** [comment]
[Update] This was discussed at the AGWG F2F at CSUN:

The group likes this language, "Content which can be triggered via pointer hover must also be accessed via the keyboard" and the feeling was we do not want to prohibit such interaction via the keyboard (or other device) in all circumstances.

NOTE: This language change if accepted by the group will be a non-normative change to the Understanding document.

**Comment by guyhickling** [comment]
@joshueoconnor  Yes, that would be a much more acceptable alternative to the current wording (or would "accessible" instead of "accessed" be more grammatical?) 

But since this is being discussed in the Understanding document, could I also suggest adding a couple of sentences to at least discourage the practice of displaying tooltips and similar things on focus? Developers don't do it intentionally, most just aren't aware of the adverse implications it has for keyboard and screen reader users. It is just the sort of thing the WCAG is here to encourage and set standards for, so I think the Understanding document could rightly make this point.

How about:

> Where text is revealed by hovering over a button or other interactive component, the best solution for keyboard and screen reader users usually is to reveal it when the user operates the button, rather than on focus unless there is good reason for doing so. This gives keyboard and screen reader users the same choice of whether or not to reveal the hidden text that mouse users have.

**Comment by patrickhlauke** [comment]
what should authors do if activating the focused interactive control is supposed to actually trigger its primary action?

**Comment by patrickhlauke** [comment]
a note telling authors that it's bad usability to display things on focus and on hover (rather than after an explicit action by the user) in general may be more appropriate and cover both cases, as hover-based stuff can be just as distracting/problematic for other user groups (e.g. low dexterity users)

**Comment by alastc** [comment]
In the context of the understanding document, I think Patrick's suggestion is most appropriate:
> a note telling authors that it's bad usability to display things on focus and on hover (rather than after an explicit action by the user)

Not banning, not particularly encouraging an approach, but discouraging showing info on hover/focus.

That would in effect encourage separate buttons for show/hide. Does that make sense?

**Comment by alastc** [comment]
I've noted the updates agreed here in the main issue for the understandin doc: #784.

So closing this, but it's not lost, just moved across there.
